Fix a misnomer in Config::permanentlyFreeze().

MenloDorian opened the pull request:
#### 50ae87b49b3150bb49ed052e38630f4f6c628dab
<pre>
Fix a misnomer in Config::permanentlyFreeze().
<a href="https://bugs.webkit.org/show_bug.cgi?id=243980">https://bugs.webkit.org/show_bug.cgi?id=243980</a>
&lt;rdar://problem/98710591&gt;

Reviewed by Justin Michaud.

The purpose of the `true` value passed to `vm_protect` is not to
`DisallowPermissionChangesAfterThis`, but to `UpdateMaximumPermission`.
Rename the related enums accordingly to reflect this.

* Source/WTF/wtf/WTFConfig.cpp:
(WTF::Config::permanentlyFreeze):

Canonical link: <a href="https://commits.webkit.org/253481@main">https://commits.webkit.org/253481@main</a>
</pre>

justinmichaud commented on Source/WTF/wtf/WTFConfig.cpp:
```diff
@@ -109,12 +109,12 @@ void Config::permanentlyFreeze()
     enum {
-        AllowPermissionChangesAfterThis = false,
-        DisallowPermissionChangesAfterThis = true
+        DontUpdateMaximumPermission = false,
```
Should this be UpdateCurrentPermissionOnly? I am confused about the docs

MenloDorian replied:
Nope.  Prototype of `vm_protect` is:
```
kern_return_t vm_protect
(
	vm_map_t target_task,
	vm_address_t address,
	vm_size_t size,
	boolean_t set_maximum,
	vm_prot_t new_protection
);
```

`DontUpdateMaximumPermission` is accurate because it tells `vm_protect` to restrict permissions to what is specified (just `VM_PROT_READ` in this case), and to not allow permissions to ever be set to anything less restrictive after this (e.g. adding back write permissions: the maximum permission is now only read).

justinmichaud reviewed (approved):
r=me

Commit:
Fix a misnomer in Config::permanentlyFreeze().
https://bugs.webkit.org/show_bug.cgi?id=243980
<rdar://problem/98710591>

Reviewed by Justin Michaud.

The purpose of the `true` value passed to `vm_protect` is not to
`DisallowPermissionChangesAfterThis`, but to `UpdateMaximumPermission`.
Rename the related enums accordingly to reflect this.

* Source/WTF/wtf/WTFConfig.cpp:
(WTF::Config::permanentlyFreeze):

Canonical link: https://commits.webkit.org/253481@main
